state/protocol/badger/validity.go

```diff
@@ -145,10 +148,6 @@ func isValidRootSnapshot(snap protocol.Snapshot) error {
 		return fmt.Errorf("root block seal for wrong block (%x != %x)", seal.BlockID, tail.ID())
 	}
 
-	if seal.ResultID != result.ID() {
```

Comment:
moved this after the identities validation, so that identities can be checked to determine whether to throw the NetworkAddressError or just a regular error